galley/pkg/server/components/processing2.go

```diff
@@ -165,7 +165,12 @@ func (p *Processing2) Start() (err error) {
 		grpcOptions = append(grpcOptions, grpc.Creds(credentials))
 	}
 	grpc.EnableTracing = p.args.EnableGRPCTracing
-	p.grpcServer = grpc.NewServer(grpcOptions...)
+
+	if p.args.InsecureGRPC != nil {
+		p.grpcServer = p.args.InsecureGRPC
+	} else {
+		p.grpcServer = grpc.NewServer(grpcOptions...)
+	}
```

Comment:
Can you move this closer to the other grpcServer setup at line 218? I think you can merge this conditional with line 224 to simplify things a bit.

Reply:
Done